src/packaging/package.rs

```diff
@@ -11,7 +8,7 @@ pub type SymbolRef = Rc<Symbol>;
 pub struct Package {
     name: SymbolRef,
     symbols: HashMap<String, SymbolRef>,
-    accessible: HashSet<SymbolRef>,
+    names: HashMap<SymbolRef, String>,
```

Comment:
This seems unnecessary. Symbols should be able to cheaply provide their own names.

Reply:
Is this intended to support aliases/local-nicknames? My comment above was just considering the already-implemented basic behavior.

Reply:
No, I implemented this memoization as a workaround for the fact that `Symbol::name` can error:
```rust
pub fn name(&self) -> Result<&String> {
    if self.is_root() {
        bail!("The root symbol doesn't have a name")
    } else if self.is_root_keyword() {
        bail!("The root keyword doesn't have a name")
    } else {
        Ok(&self.path[self.path.len() - 1])
    }
}
```